src/core/trtserver.cc

```diff
@@ -570,6 +570,31 @@ TRTSERVER_SharedMemoryBlockDelete(
   return nullptr;  // Success
 }
 
+TRTSERVER_Error*
+TRTSERVER_SharedMemoryBlockMemoryType(
+    TRTSERVER_SharedMemoryBlock* shared_memory_block,
+    TRTSERVER_Memory_Type* memory_type)
+{
+  TrtServerSharedMemoryBlock* lsmb =
+      reinterpret_cast<TrtServerSharedMemoryBlock*>(shared_memory_block);
+  *memory_type = lsmb->Type();
+  return nullptr;  // Success
+}
+
+TRTSERVER_Error*
+TRTSERVER_SharedMemoryBlockMemoryTypeId(
+    TRTSERVER_SharedMemoryBlock* shared_memory_block, int64_t* memory_type_id)
+{
+  TrtServerSharedMemoryBlock* lsmb =
+      reinterpret_cast<TrtServerSharedMemoryBlock*>(shared_memory_block);
+#ifdef TRTIS_ENABLE_GPU
+  *memory_type_id = lsmb->DeviceId();
+#else
+  *memory_type_id = 0;
+#endif             // TRTIS_ENABLE_GPU
```

Comment:
Is `DeviceId()` not defined if `TRTIS_ENABLE_GPU=OFF`? If it is defined, you don't need `#ifdef` directive.

Reply:
Yes, `DeviceId()` is not defined when `TRTIS_ENABLE_GPU_=OFF`